src/model/transaction/mergeBlockDataForContentState.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule mergeBlockDataForContentState
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+var Immutable = require('immutable');
+
+var {Map} = Immutable;
+
+import type ContentState from 'ContentState';
+import type SelectionState from 'SelectionState';
+
+function setBlockDataForContentState(
+  contentState: ContentState,
+  selectionState: SelectionState,
+  blockData: Map,
+): ContentState {
+  var startKey = selectionState.getStartKey();
+  var endKey = selectionState.getEndKey();
+  var blockMap = contentState.getBlockMap();
+  var newBlocks = blockMap
+    .toSeq()
+    .skipUntil((_, k) => k === startKey)
+    .takeUntil((_, k) => k === endKey)
+    .concat(Immutable.Map([[endKey, blockMap.get(endKey)]]))
+    .map(block => block.merge({data: block.getData().merge(blockData)}));
+
+  return contentState.merge({
+    blockMap: blockMap.merge(newBlocks),
+    selectionBefore: selectionState,
+    selectionAfter: selectionState,
+  });
+}
+
+module.exports = setBlockDataForContentState;
```

Comment:
There's a lot of overlap between these two transaction modules...looks like some copy-pasting here. :)

Can they be combined into a single module, perhaps with a fourth parameter indicating the intended behavior? `Modifier` can continue to have two methods exposed, but they can call into the same function.

Also, would you mind adding a unit test for this transaction function?